src/chart_types/heatmap/specs/heatmap.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+
+import { ChartTypes } from '../..';
+import { ScaleType } from '../../../scales/constants';
+import { SeriesScales, Spec } from '../../../specs';
+import { SpecTypes } from '../../../specs/constants';
+import { getConnect, specComponentFactory } from '../../../state/spec_factory';
+import { AccessorFn } from '../../../utils/accessor';
+import { Color, Datum, RecursivePartial } from '../../../utils/commons';
+import { config } from '../layout/config/config';
+import { Config } from '../layout/types/config_types';
+import { Predicate } from '../utils/commons';
+
+const defaultProps = {
+  chartType: ChartTypes.Heatmap,
+  specType: SpecTypes.Series,
+  data: [],
+  colors: ['red', 'yellow', 'green'],
+  colorScale: ScaleType.Linear,
+  xAccessor: ({ x }: { x: string | number }) => x,
+  yAccessor: ({ y }: { y: string | number }) => y,
+  xScaleType: ScaleType.Ordinal,
+  valueAccessor: ({ value }: { value: string | number }) => value,
+  valueFormatter: (value: number) => `${value}`,
+  xSortPredicate: Predicate.AlphaAsc,
+  ySortPredicate: Predicate.AlphaAsc,
+  config,
+};
+
+export type HeatmapScaleType =
+  | typeof ScaleType.Linear
+  | typeof ScaleType.Quantile
+  | typeof ScaleType.Quantize
+  | typeof ScaleType.Threshold;
+
+/** @alpha */
+export interface HeatmapSpec extends Spec {
+  specType: typeof SpecTypes.Series;
+  chartType: typeof ChartTypes.Heatmap;
+  data: Datum[];
+  // colorDomain?: number[];
+  colorScale?: HeatmapScaleType;
+  ranges?: number[] | [number, number];
+  colors: Color[];
+  xAccessor: AccessorFn;
+  yAccessor: AccessorFn;
+  valueAccessor: AccessorFn;
```

Comment:
Is there a reason these can't be a simple, non-functional, accessor?

```suggestion
  xAccessor: Accessor | AccessorFn;
  yAccessor: Accessor | AccessorFn;
  valueAccessor: Accessor | AccessorFn;
```

Reply:
Added in https://github.com/elastic/elastic-charts/pull/831/commits/58e64ffc81260f504049dbfe0428f9e2339682eb